lib/plugins/aws/package/compile/events/msk/index.js

```diff
@@ -0,0 +1,176 @@
+          type: 'string',
+          enum: ['LATEST', 'TRIM_HORIZON'],
+        },
+        topic: {
+          type: 'string',
+        },
+      },
+      additionalProperties: false,
+      required: ['arn', 'topic'],
+    });
+  }
+
+  resolveDependsOn(funcRole) {
+    let dependsOn = 'IamRoleLambdaExecution';
+
+    if (funcRole) {
+      if (
+        // check whether the custom role is an ARN
+        typeof funcRole === 'string' &&
+        funcRole.indexOf(':') !== -1
+      ) {
+        dependsOn = [];
+      } else if (
+        // otherwise, check if we have an in-service reference to a role ARN
+        typeof funcRole === 'object' &&
+        'Fn::GetAtt' in funcRole &&
+        Array.isArray(funcRole['Fn::GetAtt']) &&
+        funcRole['Fn::GetAtt'].length === 2 &&
+        typeof funcRole['Fn::GetAtt'][0] === 'string' &&
+        typeof funcRole['Fn::GetAtt'][1] === 'string' &&
+        funcRole['Fn::GetAtt'][1] === 'Arn'
+      ) {
+        dependsOn = funcRole['Fn::GetAtt'][0];
+      } else if (
+        // otherwise, check if we have an import or parameters ref
+        typeof funcRole === 'object' &&
+        ('Fn::ImportValue' in funcRole || 'Ref' in funcRole)
+      ) {
+        dependsOn = [];
+      } else if (typeof funcRole === 'string') {
+        dependsOn = funcRole;
+      }
+    }
+
+    return dependsOn;
+  }
+
+  getMSKClusterName(eventSourceArn) {
+    if (eventSourceArn['Fn::ImportValue']) {
+      return eventSourceArn['Fn::ImportValue'];
+    } else if (eventSourceArn.Ref) {
+      return eventSourceArn.Ref;
+    }
+
+    return eventSourceArn.split('/')[1];
+  }
+
+  compileMSKEvents() {
+    this.serverless.service.getAllFunctions().forEach(functionName => {
+      const functionObj = this.serverless.service.getFunction(functionName);
+      const cfTemplate = this.serverless.service.provider.compiledCloudFormationTemplate;
+
+      if (functionObj.events) {
+        // It is required to add the following statement in order to be able to connect to MSK cluster
+        const ec2Statement = {
+          Effect: 'Allow',
+          Action: [
+            'ec2:CreateNetworkInterface',
+            'ec2:DescribeNetworkInterfaces',
+            'ec2:DescribeVpcs',
+            'ec2:DeleteNetworkInterface',
+            'ec2:DescribeSubnets',
+            'ec2:DescribeSecurityGroups',
+          ],
+          Resource: '*',
+        };
+        const mskStatement = {
+          Effect: 'Allow',
+          Action: ['kafka:DescribeCluster', 'kafka:GetBootstrapBrokers'],
+          Resource: [],
+        };
+
+        functionObj.events.forEach(event => {
+          if (event.msk) {
+            const EventSourceArn = event.msk.arn;
+            const Topic = event.msk.topic;
+            const BatchSize = event.msk.batchSize || 100;
+            const Enabled = event.msk.enabled != null ? event.msk.enabled : true;
+            const StartingPosition = event.msk.startingPosition || 'TRIM_HORIZON';
```

Comment:
Minor style suggestion: In JS convention is to use capitalized var names only for function constructors.

Therefore it'll be less confusing to use not capitalized var names, although I know it'll get more repetitive at resource configuration.

(Also we already do `dependsOn` instead of `DependsOn`)

Unfortunately this is hard to guard by lint rules, as in most cases linter can't be sure if we deal with function constructor or not.

Reply:
That makes perfect sense - I was trying to follow the convention I found in similar places (`stream` or `sqs` events) but I totally agree - changed :+1: